unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/FriendsController/FriendsController.cs

```diff
@@ -256,11 +271,37 @@ public async UniTask<FriendRequest> RejectFriendshipAsync(string friendRequestId
         public bool IsFriend(string userId) =>
             friends.ContainsKey(userId) && friends[userId].friendshipStatus == FriendshipStatus.FRIEND;
 
-        public void RemoveFriend(string friendId) =>
+        public void RemoveFriend(string friendId)
+        {
+            if (useSocialApiBridge)
+            {
+                // TODO: try await Call for social api bridge
+                InternalRemoveFriend(friendId);
+                return;
+            }
+
             apiBridge.RemoveFriend(friendId);
+        }
 
         public async UniTask<string[]> GetFriendsAsync(int limit, int skip, CancellationToken cancellationToken = default)
         {
+            if (useSocialApiBridge)
+            {
+                int startIndex = skip;
+                int endIndex = Math.Min(skip + limit, friendsSortedByName.Count) - 1;
+                int arraySize = endIndex - startIndex + 1;
+
+                var friendIds = new string[arraySize];
+
+                for (int i = startIndex; i <= endIndex; i++)
+                {
+                    string key = friendsSortedByName.Keys[i];
+                    friendIds.SetValue(friendsSortedByName[key].userId, i - startIndex);
```

Comment:
`SetValue` API belongs to a non-generic `Array`, you should avoid using it in favor of the generic indexer